common/src/main/java/org/opensearch/ml/common/transport/search/MLSearchActionRequest.java

```diff
@@ -0,0 +1,101 @@
+    String tenantId;
+
+    /**
+     * Constructor for building an MLSearchActionRequest.
+     *
+     * @param searchRequest The original {@link SearchRequest} to be wrapped.
+     * @param tenantId The tenant ID associated with the request.
+     */
+    @Builder
+    public MLSearchActionRequest(SearchRequest searchRequest, String tenantId) {
+        this.searchRequest = searchRequest;
+        this.tenantId = tenantId;
+    }
+
+    /**
+     * Deserializes an {@link MLSearchActionRequest} from a {@link StreamInput}.
+     *
+     * @param input The stream input to read from.
+     * @throws IOException If an I/O error occurs during deserialization.
+     */
+    public MLSearchActionRequest(StreamInput input) throws IOException {
+        super(input);
+        Version streamInputVersion = input.getVersion();
+        if (input.readBoolean()) {
+            searchRequest = new SearchRequest(input);
+        }
+        this.tenantId = streamInputVersion.onOrAfter(VERSION_2_19_0) ? input.readOptionalString() : null;
+    }
+
+    /**
+     * Serializes this {@link MLSearchActionRequest} to a {@link StreamOutput}.
+     *
+     * @param output The stream output to write to.
+     * @throws IOException If an I/O error occurs during serialization.
+     */
+    @Override
+    public void writeTo(StreamOutput output) throws IOException {
+        super.writeTo(output);
+        Version streamOutputVersion = output.getVersion();
+        if (searchRequest != null) {
+            output.writeBoolean(true); // user exists
+            searchRequest.writeTo(output);
+        } else {
+            output.writeBoolean(false); // user does not exist
+        }
+        if (streamOutputVersion.onOrAfter(VERSION_2_19_0)) {
+            output.writeOptionalString(tenantId);
+        }
+    }
+
+    /**
+     * Converts a generic {@link ActionRequest} into an {@link MLSearchActionRequest}.
+     * This is useful when handling requests that may need to be converted for compatibility.
+     *
+     * @param actionRequest The original {@link ActionRequest}.
+     * @return The converted {@link MLSearchActionRequest}.
+     * @throws UncheckedIOException If the conversion fails due to an I/O error.
+     */
+    public static MLSearchActionRequest fromActionRequest(ActionRequest actionRequest) {
+        if (actionRequest instanceof MLSearchActionRequest) {
+            return (MLSearchActionRequest) actionRequest;
+        }
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(); OutputStreamStreamOutput osso = new OutputStreamStreamOutput(baos)) {
+            actionRequest.writeTo(osso);
+            try (StreamInput input = new InputStreamStreamInput(new ByteArrayInputStream(baos.toByteArray()))) {
+                return new MLSearchActionRequest(input);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException("failed to parse ActionRequest into MLSearchActionRequest", e);
```

Comment:
In this exception handling, I think IllegalArgumentException is more proper in this case

The method is expecting an ActionRequest that can be converted to MLSearchActionRequest.
If the conversion fails, it's likely because the input ActionRequest is not of the expected type or format.
This is more of a logical/argument error than an I/O error.

I think if using IllegalArgumentException, it provides clearer feedback to the caller about what went wrong.

Reply:
This is literally the same method we use for all other request classes. [Example](https://github.com/opensearch-project/ml-commons/blob/06a6021ffe5bc68d27db576895561831b7dc7aaa/common/src/main/java/org/opensearch/ml/common/transport/config/MLConfigGetRequest.java#L57-L70)